runtime/vm/jfr.cpp

```diff
@@ -1016,11 +1024,11 @@ jfrThreadContextSwitchRate(J9VMThread *currentThread)
 
 			initializeEventFields(currentThread, (J9JFREvent *)jfrEvent, J9JFR_EVENT_TYPE_THREAD_CONTEXT_SWITCH_RATE);
 
-			if (-1 == jfrState->prevContextSwitchTimestamp) {
-				jfrEvent->switchRate = 0;
+			if (-1 == jfrState->prevContextSwitchTimestamp || currentTime == jfrState->prevContextSwitchTimestamp) {
+				jfrEvent->switchRate = 0.0f;
 			} else {
 				int64_t timeDelta = currentTime - jfrState->prevContextSwitchTimestamp;
-				jfrEvent->switchRate = ((double)(switches - jfrState->prevContextSwitches) / timeDelta) * 1e9;
+				jfrEvent->switchRate = (switches - jfrState->prevContextSwitches) * 1e9f / timeDelta;
```

Comment:
is the `1e9` notation supported by compilers we use? (win cl, zOS ?)

Reply:
We already use literals of that form; e.g. https://github.com/eclipse-openj9/openj9/blob/master/runtime/gc_realtime/OSInterface.cpp#L91.